### DISCUSSION TITLE
Plans on porting coc.vim to vim9script?

### DISCUSSION THREAD

**Discussion by fifididi** [neoclide/coc.nvim#3925] [General]
Are there any plans on porting coc.vim to vim9script or would that cause compatibility issues with neovim users?

**Comment by chemzqm** [comment]
Would be worthy to rewrite some slow functions with vim9 script, PR welcome.

**Reply by trev-dev** [reply]
Are we still open to this?  Knowing what we know now means that vim9script will break compatibility for not just Neovim, but Vim users using version 8 and under.

The reason I'm asking is because I'm looking for an excuse to practice and may consider submitting PRs.

**Reply by fannheyward** [reply]
Yes, PR is welcome. For compatibility, check vim version and call separate functions, like how we did for highlighting, coc use Lua to do highlighting on nvim: 
 https://github.com/neoclide/coc.nvim/blob/628f08a0bf625f1a9872608c33638f48410d171e/autoload/coc/highlight.vim#L215-L228

**Reply by trev-dev** [reply]
I'm guessing we'll probably want some good separation between vim9scripts and the existing legacy autoload scripts, and maybe call our vim9 from global/autoloaded namespaces.  I'll play a bit this weekend.
